docs/layout-of-source-files.rst

```diff
@@ -35,6 +43,37 @@ the exact version of the compiler, so that bugfix releases are still possible.
 It is possible to specify much more complex rules for the compiler version,
 the expression follows those used by `npm <https://docs.npmjs.com/misc/semver>`_.
 
+.. index:: ! pragma, experimental
+
+.. _experimental_pragma:
+
+Experimental Pragma
+-------------------
+
+The second pragma is the experimental pragma. It can be used to enable
+features of the compiler or language that are not yet enabled by default.
+The following experimental pragmas are currently supported:
+
+
+ABIEncoderV2
+~~~~~~~~~~~~
+
+The new ABI encoder is able to encode and decode arbitrarily nested
+arrays and structs. It produces less optimal code (the optimizer
+for this part of the code is still under development) and has not
+received as much testing as the old encoder. You can activate it
+using ``pragma exprimental ABIEncoderV2;``.
+
+SMTChecker
+~~~~~~~~~~
+
+This component has to be enabled at compile-time and is
```

Comment:
Could this `compile-time` be confused with compile-time of user contracts?

Reply:
Hm, maybe - hadn't thought about that. Maybe ``This component has to be enabled when the solidity compiler is built and is therefore...`` or ``This component has to be enabled at compile-time of the solidity compiler and is therefore...`` then?